docs/semgrep-app/getting-started-with-semgrep-app.md

```diff
@@ -0,0 +1,284 @@
+---
+slug: getting-started-with-semgrep-app
+append_help_link: true
+title: Getting started with Semgrep App
+description: "Get started with Semgrep App to scan for security vulnerabilities on cloud repositories hosted on GitHub and GitLab."
+---
+
+# Scanning cloud repositories with Semgrep App
+
+Semgrep App enables you to run scans on multiple repositories by integrating with your GitHub or GitLab-hosted account. Scans are powered by **rules**. A Semgrep rule encapsulates pattern-matching logic and data-flow analysis used to find code violations, security issues, outdated libraries, and other issues.
+
+
+![Diagram of Semgrep App flow](../img/semgrep-app-diagram.png "Diagram of Semgrep App flow")
+
+
+Many rules are available from [Semgrep Registry](https://semgrep.dev/r), an open-source, community-driven repository of rules. Write your own rules to create general security checks through the Registry while customizing Semgrep to work for your team's specific secret-keeping and API practices.
+
+After performing a scan, use Semgrep App's Rule Board feature to prevent pull requests (PRs) or merge requests (MRs) from merging until findings are resolved. This helps prevent vulnerable code from shipping to widely-accessible environments, such as production or staging servers.
+
+Semgrep App enables you to scan whole repositories on a cloud. You can try Semgrep CLI for local scans as well. For more information, see [Getting started with Semgrep CLI](https://semgrep.dev/docs/getting-started/).
+
+
+## Signing into Semgrep App
+
+Signing into Semgrep App requires either a GitHub or hosted GitLab account. Semgrep App also supports Single Sign-On (SSO) on Team or Enterprise tiers. This guide focuses on GitHub and GitLab sign-ins. See [SSO Configuration](https://semgrep.dev/docs/semgrep-app/sso/) for information on single sign-on.
+
+**Prerequisites:**
+
+* A GitHub or GitLab SaaS account.
+* At least one repository associated with the account.
+
+
+### Signing in with GitHub
+
+To sign into Semgrep with a GitHub account:
+
+1. Click the following link: [Sign into Semgrep](https://semgrep.dev/login?return_path=/manage/projects).
+2. Select **Sign in with GitHub**. You are redirected to the GitHub sign in page if you are not currently signed in.
+3. Sign in with your credentials.
+4. Click **Authorize semgrep-app**. See the GitHub documentation about [Authorizing GitHub Apps](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/authorizing-github-apps) to understand the scope of permissions requested by Semgrep.
+5. You are redirected back to Semgrep App.
+6. Click **Accept** to accept Semgrep's Terms of Service.
+7. Optional: Fill out the survey and click Complete or click Skip to omit this step.
+
+You are now signed in to Semgrep App.
+
+
+#### Permissions for GitHub

```

Comment:
As these permissions are subject to change over time, is it necessary to document them (and the sign-in) process? I wonder if there is feedback on the onboarding flow (CC @juliat @milanwilliams @adamboros) that if we addressed would deprecate the need for documentation here. Thoughts on how to balance the product being the documentation vs. having explicit external docs?

Reply:
(1) There was an external engineer that asked "why do we need api scope?"
(2) I referenced @raghavjain3 's document that was shared with some users.
(And those two were what prompted me to raise the priority on this task.)

I would prefer that the product be the documentation, but there was clearly a need.

Looking forward to more input from prod/design!

Reply:
I think if we are changing the onboarding flow, it's going to be a bigger change and we need to update the docs anyways. Also I do find it useful to have this explained as it can give clarity to users who look at the docs first and sign in later.

Reply:
(Resolving - no actionable item. Thank you.)